anyblok_furetui/furetui/space.py

```diff
@@ -23,24 +23,66 @@ class Space:
     icon_type = String()
 
     def get_path(self):
-        MRe = self.registry.FuretUI.Menu.Resource
-        MRo = self.registry.FuretUI.Menu.Root
-        query = MRe.query()
-        # link with this space
-        query = query.join(MRe.root)
-        query = query.filter(MRo.space == self)
-        # order
-        query = query.order_by(MRo.order.asc())
-        query = query.order_by(MRe.order.asc()).order_by(MRe.id.asc())
+        query = text("""
+            with recursive menu_tree as (
+                select
+                    fm.id,
+                    fm.order,
+                    0 as parent_order,
+                    fm.menu_type,
+                    fm.parent_id,
+                    fmr.label,
+                    false as default
+                from
+                    furetui_menu fm
+                    join furetui_menu_root fmr on fmr.id = fm.id
+                where
+                     fmr.space_code=:space_code
+
+                union all
+
+                select
+                    child.id,
+                    child.order,
+                    (parent.parent_order + parent.order) as parent_order,
+                    child.menu_type,
+                    child.parent_id,
+                    coalesce(node.label, resource.label) as label,
+                    coalesce(resource.default, false) as default
+                from
+                    furetui_menu as child
+                    left outer join furetui_menu_node as node
+                        on node.id = child.id
+                    left outer join furetui_menu_resource resource
+                        on resource.id = child.id
+                    join menu_tree as parent on parent.id = child.parent_id
+                )
+                select
+                     id
+                from
+                    menu_tree
+                where
+                     menu_type = 'Model.FuretUI.Menu.Resource'
+                     and "default" is :default
+                order by
+                     parent_order asc,
+                     "order" asc,
+                     id asc
+                limit 1;
```

Comment:
I'm wondering what's the best in term of maintainability between plain text sql query or using sqlalchemy expression query ?

Reply:
I try both, this query is faster, but it is not working with Oracle DB